pageserver/src/tenant/metadata.rs

```diff
@@ -175,17 +203,18 @@ impl TimelineMetadata {
     }
 
     pub fn to_bytes(&self) -> Result<Vec<u8>, SerializeError> {
-        let body_bytes = self.body.ser()?;
+        let body_bytes =
```

Comment:
For forward compatibility, can we write the old format for timelines that have not switched to aux files v2 yet?  That way we can retain the option to rollback the next release, but also not have to wait another week to roll out the write path

Reply:
The code logic to implement this is like: if aux file v2 is enabled, decode with JSON; however, we cannot know whether it's enabled before decoding the metadata body. It's kind of egg-and-chicken problem.

The alternative is to maintain two versions of timeline metadata here and do not automatically upgrade the timeline metadata version, WDYT?

Reply:
... which means that aux v2 enabled -> metadata v3, aux v2 disabled -> metadata v2

Reply:
fixed in https://github.com/neondatabase/neon/pull/7663/commits/396b9cf42b931181e4a7b1b029032b6f7b02f055